backend/src/tasks/pools-updater.ts

```diff
@@ -105,35 +111,45 @@ class PoolsUpdater {
   /**
    * Http request wrapper
    */
-  private query(path): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const options = {
-        host: 'api.github.com',
-        path: path,
-        method: 'GET',
-        headers: { 'user-agent': 'node.js' }
+  private async query(path): Promise<object | undefined> {
+    type axiosOptions = {
+      httpsAgent?: https.Agent;
+    }
+    const setDelay = (secs: number = 1): Promise<void> => new Promise(resolve => setTimeout(() => resolve(), secs * 1000));
+    const axiosOptions: axiosOptions = {};
+    let retry = 0;
+
+    if (config.SOCKS5PROXY.ENABLED) {
+      const socksOptions: any = {
+        agentOptions: {
+          keepAlive: true,
+        },
+        hostname: config.SOCKS5PROXY.HOST,
+        port: config.SOCKS5PROXY.PORT
       };
 
-      logger.debug('Querying: api.github.com' + path);
-
-      const request = https.get(options, (response) => {
-        const chunks_of_data: any[] = [];
-        response.on('data', (fragments) => {
-          chunks_of_data.push(fragments);
-        });
-        response.on('end', () => {
-          resolve(JSON.parse(Buffer.concat(chunks_of_data).toString()));
-        });
-        response.on('error', (error) => {
-          reject(error);
-        });
-      });
-
-      request.on('error', (error) => {
-        logger.err('Github API query failed. Reason: '  + error);
-        reject(error);
-      });
-    });
+      if (config.SOCKS5PROXY.USERNAME && config.SOCKS5PROXY.PASSWORD) {
+        socksOptions.username = config.SOCKS5PROXY.USERNAME;
+        socksOptions.password = config.SOCKS5PROXY.PASSWORD;
+      }
+
+      axiosOptions.httpsAgent = new SocksProxyAgent(socksOptions);
+    }
+
+    while(retry < 5) {
```

Comment:
Should we expose this value to `config.json`?

Reply:
Yeah that would be great